### PR TITLE
mgr/cephadm: upgrade: use check_mon_command

### PR DESCRIPTION
**Pull request by sebastian-philipp**
Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by mgfritch** [comment]
jenkins test make check

**Comment by sebastian-philipp** [comment]
```
2020-05-15T12:10:04.238 INFO:ceph.mgr.x.smithi162.stdout:May 15 12:10:03 smithi162 bash[22411]: debug 2020-05-15T12:10:03.873+0000 7f2344aac700 -1 cephadm.serve:
2020-05-15T12:10:04.239 INFO:ceph.mgr.x.smithi162.stdout:May 15 12:10:03 smithi162 bash[22411]: debug 2020-05-15T12:10:03.873+0000 7f2344aac700 -1 Traceback (most recent call last):
2020-05-15T12:10:04.239 INFO:ceph.mgr.x.smithi162.stdout:May 15 12:10:03 smithi162 bash[22411]:   File "/usr/share/ceph/mgr/cephadm/module.py", line 499, in serve
2020-05-15T12:10:04.239 INFO:ceph.mgr.x.smithi162.stdout:May 15 12:10:03 smithi162 bash[22411]:     if self.upgrade.continue_upgrade():
2020-05-15T12:10:04.239 INFO:ceph.mgr.x.smithi162.stdout:May 15 12:10:03 smithi162 bash[22411]:   File "/usr/share/ceph/mgr/cephadm/upgrade.py", line 118, in continue_upgrade
2020-05-15T12:10:04.239 INFO:ceph.mgr.x.smithi162.stdout:May 15 12:10:03 smithi162 bash[22411]:     self._do_upgrade()
2020-05-15T12:10:04.240 INFO:ceph.mgr.x.smithi162.stdout:May 15 12:10:03 smithi162 bash[22411]:   File "/usr/share/ceph/mgr/cephadm/upgrade.py", line 338, in _do_upgrade
2020-05-15T12:10:04.240 INFO:ceph.mgr.x.smithi162.stdout:May 15 12:10:03 smithi162 bash[22411]:     'who': daemon_type,
2020-05-15T12:10:04.240 INFO:ceph.mgr.x.smithi162.stdout:May 15 12:10:03 smithi162 bash[22411]:   File "/usr/share/ceph/mgr/mgr_module.py", line 1096, in check_mon_command
2020-05-15T12:10:04.240 INFO:ceph.mgr.x.smithi162.stdout:May 15 12:10:03 smithi162 bash[22411]:     raise MonCommandFailed(f'{cmd_dict["prefix"]} failed: {r.stderr}')
2020-05-15T12:10:04.240 INFO:ceph.mgr.x.smithi162.stdout:May 15 12:10:03 smithi162 bash[22411]: mgr_module.MonCommandFailed: config set failed: unrecognized config target 'crash'
```

File: src/pybind/mgr/cephadm/upgrade.py [hunk 1]
```diff
@@ -331,11 +331,11 @@ def _do_upgrade(self):
             if image_settings.get(daemon_type) != target_name:
                 logger.info('Upgrade: Setting container_image for all %s...' %
                               daemon_type)
-                ret, out, err = self.mgr.mon_command({
+                ret, out, err = self.mgr.check_mon_command({
                     'prefix': 'config set',
                     'name': 'container_image',
                     'value': target_name,
-                    'who': daemon_type,
+                    'who': name_to_config_section(daemon_type),
```
Comment (sebastian-philipp) [inline src/pybind/mgr/cephadm/upgrade.py:338]: now also fixes a bug

**Comment by sebastian-philipp** [comment]
```
2020-05-18T09:56:52.237 INFO:ceph.mgr.x.smithi138.stdout:May 18 09:56:51 smithi138 bash[6331]: debug 2020-05-18T09:56:51.845+0000 7f4b787aa700 -1 log_channel(cluster) log [ERR] : Unhandled exception from module 'cephadm' while running on mgr.x: osd ok-to-stop fa
2020-05-18T09:56:52.237 INFO:ceph.mgr.x.smithi138.stdout:May 18 09:56:51 smithi138 bash[6331]: debug 2020-05-18T09:56:51.845+0000 7f4b787aa700 -1 cephadm.serve:
2020-05-18T09:56:52.237 INFO:ceph.mgr.x.smithi138.stdout:May 18 09:56:51 smithi138 bash[6331]: debug 2020-05-18T09:56:51.845+0000 7f4b787aa700 -1 Traceback (most recent call last):
2020-05-18T09:56:52.238 INFO:ceph.mgr.x.smithi138.stdout:May 18 09:56:51 smithi138 bash[6331]:   File "/usr/share/ceph/mgr/cephadm/module.py", line 500, in serve
2020-05-18T09:56:52.238 INFO:ceph.mgr.x.smithi138.stdout:May 18 09:56:51 smithi138 bash[6331]:     if self.upgrade.continue_upgrade():
2020-05-18T09:56:52.238 INFO:ceph.mgr.x.smithi138.stdout:May 18 09:56:51 smithi138 bash[6331]:   File "/usr/share/ceph/mgr/cephadm/upgrade.py", line 118, in continue_upgrade
2020-05-18T09:56:52.238 INFO:ceph.mgr.x.smithi138.stdout:May 18 09:56:51 smithi138 bash[6331]:     self._do_upgrade()
2020-05-18T09:56:52.238 INFO:ceph.mgr.x.smithi138.stdout:May 18 09:56:51 smithi138 bash[6331]:   File "/usr/share/ceph/mgr/cephadm/upgrade.py", line 270, in _do_upgrade
2020-05-18T09:56:52.239 INFO:ceph.mgr.x.smithi138.stdout:May 18 09:56:51 smithi138 bash[6331]:     if not self._wait_for_ok_to_stop(d):
2020-05-18T09:56:52.239 INFO:ceph.mgr.x.smithi138.stdout:May 18 09:56:51 smithi138 bash[6331]:   File "/usr/share/ceph/mgr/cephadm/upgrade.py", line 132, in _wait_for_ok_to_stop
2020-05-18T09:56:52.239 INFO:ceph.mgr.x.smithi138.stdout:May 18 09:56:51 smithi138 bash[6331]:     'ids': [s.daemon_id],
2020-05-18T09:56:52.239 INFO:ceph.mgr.x.smithi138.stdout:May 18 09:56:51 smithi138 bash[6331]:   File "/usr/share/ceph/mgr/mgr_module.py", line 1096, in check_mon_command
2020-05-18T09:56:52.239 INFO:ceph.mgr.x.smithi138.stdout:May 18 09:56:51 smithi138 bash[6331]:     raise MonCommandFailed(f'{cmd_dict["prefix"]} failed: {r.stderr}')
2020-05-18T09:56:52.240 INFO:ceph.mgr.x.smithi138.stdout:May 18 09:56:51 smithi138 bash[6331]: mgr_module.MonCommandFailed: osd ok-to-stop failed: 1 PGs are already too degraded, would become too degraded or might become unavailable
```

**Comment by sebastian-philipp** [comment]
susccess http://pulpito.ceph.com/swagner-2020-05-18_15:53:54-rados-wip-swagner-testing-2020-05-18-1513-distro-basic-smithi/

**Pull request merged**